packages/transactions/src/new-compile-transaction.ts

```diff
@@ -0,0 +1,39 @@
+import { Address } from '@solana/addresses';
+import { SignatureBytes } from '@solana/keys';
+
+import { CompilableTransaction } from './compilable-transaction';
+import { compileMessage } from './message';
+import { getCompiledMessageEncoder } from './serializers/message';
+
+type TypedArrayMutableProperties = 'copyWithin' | 'fill' | 'reverse' | 'set' | 'sort';
+interface ReadonlyUint8Array extends Omit<Uint8Array, TypedArrayMutableProperties> {
+    readonly [n: number]: number;
+}
+
+export type TransactionMessageBytes = ReadonlyUint8Array & { readonly __brand: unique symbol };
+export type OrderedMap<K extends string, V> = Record<K, V>;
+
+export type NewTransaction = Readonly<{
+    messageBytes: TransactionMessageBytes;
+    signatures: OrderedMap<Address, SignatureBytes | null>;
+}>;
+
+export function compileTransactionMessage(transactionMessage: CompilableTransaction): NewTransaction {
+    const compiledMessage = compileMessage(transactionMessage);
```

Comment:
We should probably try and find a better name to distinguish these two functions: `compileTransactionMessage` and `compileMessage`.

Maybe `compileTransactionMessage` could become `compileTransaction` and `compileMessage` could become `compileTransactionMessage`.

Reply:
Okay cool I changed these to `compileTransaction` and `compileTransactionMessage` - thanks!